storageconfig: cleanup and move StoreSpec

RaduBerinde opened the pull request:
#### storageconfig: clean up SizeSpec

This change renames SizeSpec to Size (everything in `storageconfig` is
a "spec"); cleans up the interface of Size; and separates the
pflag-specific code to a wrapper inside the `cli` package.

Epic: none
Release note: None

#### storageconfig: move provisioned rate

Move `base.ProvisionedRateSpec` to `storageconfig.ProvisionedRate`.

Epic: none
Release note: None

#### storageconfig: move StoreSpec

Epic: none
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/148338)
<!-- Reviewable:end -->

RaduBerinde committed:
storageconfig: clean up SizeSpec

This change renames SizeSpec to Size (everything in `storageconfig` is
a "spec"); cleans up the interface of Size; and separates the
pflag-specific code to a wrapper inside the `cli` package.

Epic: none
Release note: None

RaduBerinde committed:
storageconfig: move StoreSpec

Note that the command-line (pflag) adapters are left out of
storageconfig, as those will in time be deprecated.

Epic: none
Release note: None

RaduBerinde committed:
storageconfig: move provisioned rate

Move `base.ProvisionedRateSpec` to `storageconfig.ProvisionedRate`.

Epic: none
Release note: None

jbowens reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 21 of 21 files at r1, 6 of 6 files at r2, 17 of 17 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/148338#-:-OTISV6CCkshFKIc_UIH:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @alyshanjahani-crl, @golgeek, and @herkolategan)



<!-- Sent from Reviewable.io -->
